projecttacoma/fqm-testify
# Multi group measures

# Summary
This PR adds support for multiple group measures in the MeasureHighlightingPanel.

## New Behavior
Before, fqm-testify only supported showing measure logic highlighting for one group of a patient; however, we have more recently come across multiple group measures such as CMS347 that has 4 groups. In the MeasureHighlightingPanel, there are tabs at the top for each of the groups (only 1 tab if it is not a multiple group measure). Each tab contains a PopulationComparisonTable and Measure highlighting for that patient and group.

## Code Changes
- `TestCaseEditor.tsx` - adds Mantine Tabs component for each group
- `MeasureHighlightingPanel.tsx` - updated the HTML to only return the HTML of a group on a patient, not all of the groups
- `PopulationComparisonTable.tsx` - adds state variable to the component so that it can be properly reused and takes in a detailed result (group result) rather than a patientId
- `PopulationComparisonTable.test.tsx` - updated unit test

# Testing Guidance
- `npm run check`
- `npm run dev` and navigate to `http://localhost:3000`
- Make sure that the app still works with single group measures (should have one tab for one group)
- Upload CMS347 make sure that the tabs work and that they display the correct results and highlighting for a patient and group
